### PR TITLE
[FIX] fetchmail: mails not being processed

### PR DESCRIPTION
**Pull request by jjscarafia**
Prove of concept, not sure why but we're getting mail read but not processed in odoo. With this change we try to ensure mail being stored before marking mail as read

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by dennybiasiolli — COMMENTED** [review]
Nice fix, but it leaves another bug:
if `message_process` from line 170 raise an error, the message will be marked as SEEN and won't be processed anymore. Please take a look at #58500 😉

File: addons/fetchmail/models/fetchmail.py [hunk 1]
```diff
@@ -169,14 +169,18 @@ def fetch_mail(self):
...
                         except Exception:
-                            _logger.info('Failed to process mail from %s server %s.', server.server_type, server.name, exc_info=True)
-                            failed += 1
-                        imap_server.store(num, '+FLAGS', '\\Seen')
+                            _logger.warning('Failed to process mail from %s server %s.', server.server_type, server.name, exc_info=True)
                         self._cr.commit()
-                        count += 1
+                        if res_id:
+                            # imap_server.store(num, '+FLAGS', '\\Seen')
+                            imap_server.store(num, '+FLAGS', '\\Seen \\Flagged')
```
Comment (dennybiasiolli) [inline addons/fetchmail/models/fetchmail.py:176]: You sure about that?

According to [RFC](https://tools.ietf.org/html/rfc3501#section-2.3.2)

> Message is "flagged" for urgent/special attention

so I think it's correct, but how does it appear on webmail clients?

Comment (dennybiasiolli) [inline addons/fetchmail/models/fetchmail.py:176]: I think the message should be flagged only in case of exceptions, not always

**Review by dennybiasiolli — CHANGES_REQUESTED** [review]
If `res_id` we can use `imap_server.store(num, '+FLAGS', '\\Seen')`, setting the flag `\Flagged` should be done only in case of exceptions during `message_process`

File: addons/fetchmail/models/fetchmail.py (hunk 1 above)
Comment (jjscarafia) [reply addons/fetchmail/models/fetchmail.py:176]: Hi, it's just a test. It seems that our emails are being read by someone else than this main process. We've change mail password and still happening, this is just a test to see if this is the method reading all the emails

**Comment by jjscarafia** [comment]
@dennybiasiolli It seems that in our side the issue was with a dangling container accessing same database :(
Closing for now..

**Pull request closed without merging**